pkg/apiserver/resthandler.go

```diff
@@ -470,11 +470,11 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
 
-		// PATCH requires same permission as UPDATE
-		if admit.Handles(admission.Update) {
+		if admit != nil && admit.Handles(admission.Patch) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(obj, scope.Kind.GroupKind(), namespace, name, scope.Resource.GroupResource(), scope.Subresource, admission.Update, userInfo))
+			// TODO this record should include the patch itself.  Right now we can't do that because patches are not runtime.Objects
```

Comment:
> Sorry for being dense here, but I don't understand how this line adds any security; doesn't everything call this AND also call the updateAdmit function that is defined below? And if yes, what is the benefit of having this "Patch" call here?

Its the only spot where you can currently inspect that "user A isn't mutating field 2".  Without both old and new you can't make that determination on an update.  I wasn't prepared to say that "all downstream bits must give up that power".

With a full authorizer, that is easy to express as: "user A can patch, but not update" and "user A can't mutate field 2".